CommonMark/Parser/BlockMethods.cs

```diff
@@ -92,16 +93,355 @@ private static void BreakOutOfLists(ref Block blockRef, LineInfo line)
+
+                    row.StringContent = new StringContent();
+                    row.StringContent.Append(asStr, offset, row.SourceLength);
+
+                    if (table.LastChild == null)
+                    {
+                        table.FirstChild = row;
+                        table.LastChild = row;
+                    }
+                    else
+                    {
+                        table.LastChild.NextSibling = row;
+                        table.LastChild = row;
+                    }
+
+                    MakeTableCells(row, sb);
+                    row.IsOpen = false;
+                }
+
+                offset += lineLength;
+            }
+        }
+
+        static bool TryMakeTable(Block b, LineInfo line, CommonMarkSettings settings)
+        {
+            if ((settings.AdditionalFeatures & CommonMarkAdditionalFeatures.GithubStyleTables) == 0) return false;
+
+            var asStr = b.StringContent.ToString();
+            var lines = asStr.Split('\n');
+
+            if (lines.Length < 2) return false;
+
+            var sb = new StringBuilder();
+
+            var columnsLine = ParseTableLine(lines[0], sb);
+            if (columnsLine.Count == 1) return false;
+
+            var headerLine = ParseTableLine(lines[1], sb);
+            if (headerLine.Count == 1) return false;
+
+            var headerAlignment = new List<TableHeaderAlignment>();
+
+            foreach (var headerPart in headerLine)
+            {
+                var trimmed = headerPart.Trim();
+                if (trimmed.Length < 3) return false;
+
+                var validateFrom = 0;
+                var startsWithColon = trimmed[validateFrom] == ':';
+                if (startsWithColon) validateFrom++;
+
+                var validateTo = trimmed.Length - 1;
+                var endsWithColon = trimmed[validateTo] == ':';
+                if (endsWithColon) validateTo--;
+
+                for (var i = validateFrom; i <= validateTo; i++)
+                {
+                    // don't check for escapes, they don't count in header
+                    if (trimmed[i] != '-') return false;
+                }
+
+                if (!startsWithColon && !endsWithColon)
+                {
+                    headerAlignment.Add(TableHeaderAlignment.None);
+                    continue;
+                }
+
+                if (startsWithColon && endsWithColon)
+                {
+                    headerAlignment.Add(TableHeaderAlignment.Center);
+                    continue;
+                }
+
+                if (startsWithColon)
+                {
+                    headerAlignment.Add(TableHeaderAlignment.Left);
+                }
+
+                if (endsWithColon)
+                {
+                    headerAlignment.Add(TableHeaderAlignment.Right);
+                }
+            }
+
+            while (columnsLine.Count > 0 && string.IsNullOrWhiteSpace(columnsLine[0])) columnsLine.RemoveAt(0);
```

Comment:
IsNullOrWhiteSpace cannot be used.
shouldn't this check be done in ParseTableLine method?